test/e2e/aks_clusterclass.go

```diff
@@ -66,14 +66,16 @@ func AKSClusterClassSpec(ctx context.Context, inputGetter func() AKSClusterClass
 	By("Editing the AzureManagedMachinePoolTemplate to change the scale down mode")
 	ammpt := &infrav1.AzureManagedMachinePoolTemplate{}
 
-	// TODO: We are hard-coding the ammpt name suffix since to get the name from
-	// the ClusterClass, we have to upgrade CAPI to v1.6.0. Fix this once CAPI
-	// v1.6.0 is released.
-	// CAPI v1.6.0 Bump PR: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4182
+	clusterClass := &clusterv1.ClusterClass{}
+	err = mgmtClient.Get(ctx, types.NamespacedName{
+		Namespace: input.Cluster.Namespace,
+		Name:      "default",
+	}, clusterClass)
+
 	Eventually(func(g Gomega) {
 		err = mgmtClient.Get(ctx, types.NamespacedName{
-			Namespace: input.Cluster.Namespace,
-			Name:      input.Cluster.Name + "-pool0",
+			Namespace: clusterClass.Spec.Workers.MachinePools[0].Template.Infrastructure.Ref.Namespace,
```

Comment:
could we range over the MachinePools and use the index instead of hardcoding to 0 here?